### ISSUE TITLE
[BUG] Caller should not connect on Rendezvous

### ISSUE THREAD

**Issue by bpolic** [Haivision/srt#2555]
**Describe the bug**
Caller can connect on Rendezvous

Linux machine 1 IP: yy.yy.yy.yy
Linux machine 2 IP: xx.xx.xx.xx

**To Reproduce**
1. Use 2 Linux machines
2. Build a SRT Library
3. Run caller on first maxhine: `./srt-xtransmit receive "srt://xx.xx.xx.xx:8200?latency=500&bind=:8200" -v`
4. On another machine run rendezvous: `./srt-xtransmit generate "srt://yy.yy.yy.yy:8200?latency=500&mode=rendezvous&bind=:8200" -v --duration 20 --sendrate 5Mbps`
5. Current behavior: The connection has been established, and the data has been sent.

"Rendezvous-point received INDUCTION handshake (expected WAVEAHAND). Rejecting.
 SOCKET::SRT 0x15F1BCC7 ASYNC Connected to srt://xx.xx.xx.xx:8200
17:20:34.310462 [I] PACER sendrate 5000000 bps (inter send interval 2105 us)"

**Expected behavior**
Caller should not connect on Rendezvous

**Desktop (please provide the following information):**
 - OS: Linux
 - SRT Version 1.5.1

**Comment by maxsharabayko** [comment]
This issue seems to be resolved in the latest master (6d620961b52549916d8b8370503120dda9ec2d18).

```shell
./bin/srt-xtransmit receive "srt://127.0.0.1:4200?latency=500&bind=:5200" -v
10:30:57.456514 [I] SOCKET::SRT srt://127.0.0.1:4200: bound to ':5200'.
10:30:57.457216 [D] SOCKET::SRT 0x160BDB ASYNC Connecting to srt://127.0.0.1:4200
10:30:58.280427/*E:SRT.cn: @1444827: HS VERSION=5 but no handshake extension found!
10:30:58.280492/!W:SRT.cn: @1444827: processAsyncConnectRequest: REJECT reported from HS processing: Rogue peer or incorrect parameters - not processing further
10:30:58.283700 [D] SOCKET::SRT 0x160BDB connect failed. ERROR: Rogue peer or incorrect parameters.
10:30:58.283848 [D] SOCKET::SRT 0x160BDB Releasing epolls before closing
10:30:58.283888 [D] SOCKET::SRT 0x160BDB Closing
10:30:58.284024 [W] CONN connect failed: Rogue peer or incorrect parameters
```

```shell
./bin/srt-xtransmit generate "srt://127.0.0.1:5200?bind=:4200&mode=rendezvous" --duration 20s --sendrate 1Mbps -v
10:30:58.278782 [I] SOCKET::SRT srt://127.0.0.1:5200: bound to ':4200'.
10:30:58.279465 [D] SOCKET::SRT 0x2471F434 ASYNC Connecting to srt://127.0.0.1:5200
10:30:58.280608/*E:SRT.cn: @611447860: processConnectResponse: CONFUSED: expected UMSG_HANDSHAKE as connection not yet established, got: shutdown
10:30:58.280646/!W:SRT.cn: AsyncOrRND: PACKET NOT HANDSHAKE - re-requesting handshake from peer
10:31:28.296288 [D] SOCKET::SRT 0x2471F434 connect failed. ERROR: Rogue peer or incorrect parameters.
10:31:28.296491 [D] SOCKET::SRT 0x2471F434 Releasing epolls before closing
10:31:28.296549 [D] SOCKET::SRT 0x2471F434 Closing
10:31:28.296715 [W] CONN connect failed: Rogue peer or incorrect parameters
```

**Comment by maxsharabayko** [comment]
Looks like fixed by #2548.

**Comment by ethouris** [comment]
Max, but your test isn't doing the same thing as the report. The report speaks about using the same port number for binding and connecting.
